e2e/nx-plugin-e2e/tests/init.e2e.test.ts

```diff
@@ -1,25 +1,80 @@
-import { expect } from 'vitest';
+import { Tree } from '@nx/devkit';
+import { rm } from 'node:fs/promises';
+import { join } from 'node:path';
+import { afterEach, expect } from 'vitest';
+import {
+  generateWorkspaceAndProject,
+  materializeTree,
+} from '@code-pushup/test-nx-utils';
 import { removeColorCodes } from '@code-pushup/test-utils';
-import { executeProcess } from '@code-pushup/utils';
+import { distPluginPackage, executeGenerator } from '../mocks/utils';
+
+function executeInitGenerator(args: string[], cwd: string = process.cwd()) {
+  return executeGenerator(args, {
+    bin: distPluginPackage(cwd),
+    generator: 'init',
+    cwd,
+  });
+}
 
 describe('nx-plugin g init', () => {
-  it('should run init generator and execute correctly', async () => {
-    const { code, stderr, stdout } = await executeProcess({
-      command: 'npx',
-      args: ['nx', 'g', '@code-pushup/nx-plugin:init --dryRun'],
-      observer: { onStdout: console.info },
-    });
+  let tree: Tree;
+  const project = 'my-lib';
+  const baseDir = 'tmp/nx-plugin-e2e/generators/configuration';
+
+  beforeEach(async () => {
+    tree = await generateWorkspaceAndProject(project);
+  });
+
+  afterEach(async () => {
+    await rm(baseDir, { recursive: true, force: true });
+  });
+
+  it('should inform about dry run', async () => {
+    const cwd = join(baseDir, 'configure');
+    await materializeTree(tree, cwd);
+
+    const { stderr } = await executeInitGenerator([project, '--dryRun'], cwd);
```

Comment:
Here an example of the new test helper un use. We can reuse Nx helpers to generate and logic from our own generators to create dynamic test scenarios in the file system.